docs/source/resources/graphql-glossary.md

```diff
@@ -0,0 +1,240 @@
+
+<h2 id="normalization">Normalization</h2>
+<p>A technique for transforming default GraphQL responses into a specific desired format.</p>
+
+```js
+// Example of a default GraphQL response
+const response = {
+  data: {
+    getUser: {
+      __typename: 'User',
+      uid: '5a6efb94b0e8c36f99fba013',
+      email: 'john.doe@yahoo.com',
+    },
+  },
+}
+
+// After Normalization
+{
+ users: {
+    '5a6efb94b0e8c36f99fba013': {
+      uid: '5a6efb94b0e8c36f99fba013',
+      email: 'john.doe@yahoo.com'
+   }
+ }
+}
+```
+
+<h2 id="object-type">Object Type</h2>
+<p>A form of Object with a type specifier that has fields that can fetch data from an API service, e.g User is an Object type.</p>
+
+```js
+type User {
+   name: String!,
+}
+```
+
+<h2 id="operation">Operation</h2>
+<p>A single query, mutation, or subscription that can be interpreted by a GraphQL execution engine.</p>
+
+<h2 id="operation-name">Operation name</h2>
+<p>A name for a single query, mutation, or subscription. Identifying a query or mutation by name is very useful for logging and debugging when something goes wrong in a GraphQL server.</p>
+
+<h2 id="partial-query-caching">Partial query caching</h2>
+<p>A technique for caching inputs to GraphQL queries. This type of caching ensures that if the query is slightly different but with the same inputs, those inputs can simply be retrieved from the cache instead of fetching data again from the backend. It is implemented in Apollo Server 2 as Data Source caching.</p>
+
+<h2 id="query">Query</h2>
+<p>An operation that makes a GET request to a GraphQL service requesting for some data. It's better known as a read-only fetch operation.</p>
+
+<h2 id="query-colocation">Query colocation</h2>
+<p>A practice of placing a GraphQL query in the same location as the app component's view logic.</p>
+
+```js
+const GET_DOG_PHOTO = gql`
+ query dog($breed: String!) {
+  dog(breed: $breed) {
+    id
+    displayImage
+  }
+}`;
+
+export const queryComponent = `const DogPhoto = ({ breed }) => (
+  <Query query={GET_DOG_PHOTO} variables={{ breed }}>
+    {({ loading, error, data }) => {
+      if (loading) return null;
+      if (error) return 'Error!';
+      return (
+        <img src={data.dog.displayImage} />
+      );
+    }}
+  </Query>
+);`;
+```
+
+<h2 id="query-whitelisting">Query whitelisting</h2>
+<p>A technique for preventing unwanted attacks by maintaining a list of approved queries that are allowed in your application. Any query not present in the list that is run against the server will not be allowed. [Automatic Persisted Queries](../guides/performance.html#automatic-persisted-queries) is a tool by Apollo that enables query whitelisting and persisted queries.</p>
+
+<h2 id="resolver">Resolver</h2>
+<p>A function that connects schema fields and types to various backends. It can retrieve or write data from either an SQL, a No-SQL, graph database, a micro-service or a REST API.</p>
```

Comment:
Resolvers can do 2 things: fetch data or transform data

Reply:
When you transform data, are you not writing data? Does writing not cover creating & modifying? @JakeDawkins